rpk: Add guardrails for Tiered Storage properties

Lazin opened the pull request:
Prompt the user for confirmation if they are trying to set `cloud_storage_enable_remote_write` cluster config to 'false'. Do the same if they're trying to set topic property `redpanda.remote.write` to 'false'. This operation is potentially unsafe because the data loss might occur if the Tiered Storage was active. In both cases the message says `Warning: disabling Tiered Storage may lead to data loss. If you only want to pause Tiered Storage temporarily, use the 'cloud_storage_enable_segment_uploads' option. Abort?` and the default choice is `Yes`. So the warning suggests a safe alternative and in order to use non-safe option the user have to chose it explicitly. 

This is a part of the safe pause/resume feature - https://github.com/redpanda-data/docs/pull/1017 (docs PR).

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v25.1.x
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->
* none

Lazin committed:
rpk: Prompt user for positive confirmation

Prompt if the user is trying to set 'cloud_storage_enable_remote_write'
to 'false'. This operation is potentialy unsafe because the data loss
might occur.

The prompt can be avoided by passing '--no-confirm'. This flag is
supposed to be used in tests and automation scripts.

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

Lazin committed:
rpk: Prompt for positive confirmation

in case if the user is disabling 'redpanda.remote.write'. This is a
potentially dangerous operation that can lead to data loss. The prompt
suggest an alternaive for that (safe pause).

The '--no-confirm' flag that allows the user to bypass the prompt is
added. The flag is supposed to be used in scripts and CI.

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/63439-->
<!--b4a097d1-3d84-4a3f-8c9d-5682ff08f2cb-->
<!--2dc24b66-4091-4619-980a-1d8073d419b7-->
<!--eb086872-0663-42bf-9fb9-f8adf722640a-->
<!--1899c84d-0966-4ce9-8c4d-f62ae0e07ab1-->
<!--0a01a713-8731-4851-9a6c-b6ab4388c434-->
<!--62cbefde-495d-4e7b-a978-b35d95b9ff2e-->
<!--644637be-c796-4693-bf24-8689f59026d4-->
<!--eb27ed46-1260-444f-8d18-1365fbdaa84e-->
<!--title-->
## Retry command for [Build#63439](https://buildkite.com/redpanda/redpanda/builds/63439)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```


/ci-repeat 1
tests/rptest/tests/delete_retention_ms_test.py::DeleteRetentionMsTest.test_create_topic_configurations_cluster_defaults
tests/rptest/tests/alter_topic_configuration_test.py::AlterTopicConfiguration.test_shadow_indexing_config
tests/rptest/tests/cloud_storage_scrubber_test.py::CloudStorageScrubberTest.test_scrubber@{"cloud_storage_type":1}
tests/rptest/tests/e2e_shadow_indexing_test.py::EndToEndShadowIndexingTest.test_reset@{"cloud_storage_type":2}
tests/rptest/tests/e2e_shadow_indexing_test.py::EndToEndShadowIndexingTest.test_reset_spillover@{"cloud_storage_type":2}
tests/rptest/tests/alter_topic_configuration_test.py::ShadowIndexingGlobalConfig.test_overrides_remove
tests/rptest/tests/cloud_storage_scrubber_test.py::CloudStorageScrubberTest.test_scrubber@{"cloud_storage_type":2}
tests/rptest/tests/datalake/transactions_test.py::DatalakeTransactionTests.test_with_transactions@{"cloud_storage_type":1,"compaction":false}
tests/rptest/tests/e2e_shadow_indexing_test.py::EndToEndShadowIndexingTest.test_reset@{"cloud_storage_type":1}
tests/rptest/tests/e2e_shadow_indexing_test.py::EndToEndShadowIndexingTest.test_reset_spillover@{"cloud_storage_type":1}
tests/rptest/tests/alter_topic_configuration_test.py::ShadowIndexingGlobalConfig.test_overrides_set
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--7a13574a-e4e6-4f96-a80e-fa0c7ca7a8f9-->
<!--TEST RESULTS-->
<!--597fdd0b-431b-4007-978b-40addc78e0ed-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/63439">build#63439</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.alter_topic_configuration_test.AlterTopicConfiguration.test_shadow_indexing_config | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5a-4da9-9c4b-84f6a83239c5 | FAIL | 0/20 |
| rptest.tests.alter_topic_configuration_test.AlterTopicConfiguration.test_shadow_indexing_config | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f1e-49ac-bc16-2d86c69b137a | FAIL | 0/20 |
| rptest.tests.alter_topic_configuration_test.ShadowIndexingGlobalConfig.test_overrides_remove | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5b-4db7-9f61-b51c9031295f | FAIL | 0/20 |
| rptest.tests.alter_topic_configuration_test.ShadowIndexingGlobalConfig.test_overrides_remove | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f1f-4987-98db-400d3936e567 | FAIL | 0/20 |
| rptest.tests.alter_topic_configuration_test.ShadowIndexingGlobalConfig.test_overrides_set | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5c-414e-b841-36f5e5d96b14 | FAIL | 0/20 |
| rptest.tests.alter_topic_configuration_test.ShadowIndexingGlobalConfig.test_overrides_set | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f20-4901-97a3-2964c35b3d70 | FAIL | 0/20 |
| rptest.tests.cloud_storage_scrubber_test.CloudStorageScrubberTest.test_scrubber.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5c-414e-b841-36f5e5d96b14 | FAIL | 0/20 |
| rptest.tests.cloud_storage_scrubber_test.CloudStorageScrubberTest.test_scrubber.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f20-4901-97a3-2964c35b3d70 | FAIL | 0/20 |
| rptest.tests.cloud_storage_scrubber_test.CloudStorageScrubberTest.test_scrubber.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5a-47af-a107-9ea3c3df8d4f | FAIL | 0/20 |
| rptest.tests.cloud_storage_scrubber_test.CloudStorageScrubberTest.test_scrubber.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f1e-481f-8d6e-fe47ec5c9638 | FAIL | 0/20 |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_migrated_topic_data_integrity.transfer_leadership=True.params=.cancellation.dir.out.stage.preparing.use_alias.False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5b-4db7-9f61-b51c9031295f | FLAKY | 1/2 |
| rptest.tests.datalake.compaction_test.CompactionTest.test_compaction.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f1e-49ac-bc16-2d86c69b137a | FLAKY | 1/2 |
| rptest.tests.datalake.datalake_upgrade_test.DatalakeUpgradeTest.test_upload_through_upgrade.cloud_storage_type=CloudStorageType.S3.query_engine=QueryEngineType.SPARK | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f20-4901-97a3-2964c35b3d70 | FLAKY | 1/2 |
| rptest.tests.datalake.transactions_test.DatalakeTransactionTests.test_with_transactions.cloud_storage_type=CloudStorageType.S3.compaction=False | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5c-414e-b841-36f5e5d96b14 | FAIL | 0/1 |
| rptest.tests.delete_retention_ms_test.DeleteRetentionMsTest.test_create_topic_configurations_cluster_defaults | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5a-4da9-9c4b-84f6a83239c5 | FAIL | 0/20 |
| rptest.tests.delete_retention_ms_test.DeleteRetentionMsTest.test_create_topic_configurations_cluster_defaults | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f1f-4987-98db-400d3936e567 | FAIL | 0/20 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_reset.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5b-4db7-9f61-b51c9031295f | FAIL | 0/20 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_reset.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f1e-49ac-bc16-2d86c69b137a | FAIL | 0/20 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_reset.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5c-414e-b841-36f5e5d96b14 | FAIL | 0/20 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_reset.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f1f-4987-98db-400d3936e567 | FAIL | 0/20 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_reset_spillover.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5b-4db7-9f61-b51c9031295f | FAIL | 0/20 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_reset_spillover.cloud_storage_type=CloudStorageType.ABS | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f1e-49ac-bc16-2d86c69b137a | FAIL | 0/20 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_reset_spillover.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5c-414e-b841-36f5e5d96b14 | FAIL | 0/20 |
| rptest.tests.e2e_shadow_indexing_test.EndToEndShadowIndexingTest.test_reset_spillover.cloud_storage_type=CloudStorageType.S3 | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b465-0f1f-4987-98db-400d3936e567 | FAIL | 0/20 |
| rptest.tests.schema_registry_test.SchemaRegistryAutoAuthTest.test_schema_id_validation.protocol=AVRO.client_type=Python.validate_schema_id=True.subject_name_strategy=SubjectNameStrategyCompat.TOPIC_RECORD_NAME.payload_class=com.redpanda.A.B.C.D.NestedPayload.compression_type=CompressionTypes.ZSTD | ducktape | https://buildkite.com/redpanda/redpanda/builds/63439#0195b444-3f5a-4da9-9c4b-84f6a83239c5 | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/63977">build#63977</a></summary>


| test_id                                                                                                                                                                       | test_kind   | job_url                                                                                   | test_status   | passed   |
|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.datalake.cluster_restore_test.DatalakeClusterRestoreTest.test_restore_partition_spec.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63977#0195ecd2-34d8-4ca7-9a2a-b896915acf8c | FLAKY         | 1/2      |
| rptest.tests.datalake.throttling_test.DatalakeThrottlingTest.test_basic_throttling.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.NESSIE                     | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63977#0195ecd2-34db-49e1-af43-810c416fdb15 | FLAKY         | 1/2      |
| rptest.tests.upgrade_test.UpgradeWithWorkloadTest.test_rolling_upgrade                                                                                                        | ducktape    | https://buildkite.com/redpanda/redpanda/builds/63977#0195eccf-f71e-4593-a176-5f78928e94ce | FLAKY         | 1/2      |
</details>
<!--/body-->

r-vasquez commented on src/go/rpk/pkg/cli/cluster/config/set.go:
```diff
@@ -69,6 +70,14 @@ If an empty string is given as the value, the property is reset to its default.`
 				out.Die("invalid arguments: %v, please use one of 'rpk cluster config set <key> <value>' or 'rpk cluster config set <key>=<value>'", args)
 			}
+			// Disabling Tiered Storage requires a confirmation from the user because it may lead to data loss.
+			if key == "cloud_storage_enable_remote_write" && value == "false" {
```
LGTM, however, we should provide a way to avoid these confirmation prompts in the CLI. Similar to what we do in other commands with confirmation prompts.

Example: https://github.com/redpanda-data/redpanda/blob/dev/src/go/rpk/pkg/cli/security/role/delete.go#L71
_we use this flag name across rpk_

The same comment applies to topic config.

Lazin replied:
done

Feediver1 commented:
@Lazin Not sure what to do with the pause/resume doc PR, as this has not yet been merged. Need some guidance on whether this will be added before GA. thx

Lazin committed:
rptest: Pass --no-confirm flag

In order to disable TS the rpk user must provide positive confirmation
now. This breaks some tests. In order to avoid this the '--no-confirm'
flag is passed to the rpk.

Signed-off-by: Evgeny Lazin <4lazin@gmail.com>

Lazin commented:
@Feediver1  I'm going to merge this and backport to v25.1.x but it's not part of the RC

vbotbuildovich commented:
/backport v25.1.x
